### ISSUE TITLE
Bug in ServerSideRender where the block disappears during render.

### ISSUE THREAD

**Issue by pryley** [WordPress/gutenberg#35406]
If you change a prop (block option) during the loading stage (while the spinner is showing), the entire content of the block disappears.

## Before

https://user-images.githubusercontent.com/134939/135711766-d85154f5-c9a2-4447-aabf-70fadc8235b4.mp4

## After

https://user-images.githubusercontent.com/134939/135711774-c450a095-bacb-44ea-91b5-9f37dbf609fd.mp4

_Originally posted by @pryley in https://github.com/WordPress/gutenberg/issues/28297#issuecomment-932723947_

**Comment by fabiankaegy** [comment]
I am talking a look at this. It appears that it was introduced after the change to the preview mechanism in #28289 or #35033

**Comment by fabiankaegy** [comment]
I started looking into this.

The issue appears to be that we set the response to be `null` every time we start fetching the markup here: https://github.com/WordPress/gutenberg/blob/9e782ed0e9ef57058f0d6b3fa5e8bce6d44c3090/packages/server-side-render/src/server-side-render.js#L88-L90

The issue with that is that we us the `usePrevious` hook to keep track of the markup of the component while we re fetch the markup. Once we start a new fetch request before the previous one ressolved we push the `null` value to the `previousMarkup` and that is what causes the markup to dissapear. 

I am continuing to investigate on how we can solve this
